docs/core/classes.md

```diff
@@ -411,3 +411,98 @@ qx.Class.define("qx.test.Cat", {
   include: [qx.test.MPet, qx.test.MSleep]
 });
 ```
+
+# Delegating a class' instances' setter, getter, and delete
+
+This feature is EXPERIMENTAL. Its interface may yet change.
+
+On occasion, it is desirable to change the behavior of accessing
+properties on all instances of a class. For example, a class that has
+`setItem` and `getItem` methods for accessing indexed values within
+the class, and a `removeItem` method for deleting an item, might allow
+the user to set the 3rd (zero-relative) value with
+`instance.setItem(2, 42);`; to retrieve it with `value =
+instance.getItem(2);`; and to delete it with `instance.removeAt(2);`.
+
+For usability, the developer might like for those statements to be
+accessed using the array index operator, as such:
+
+```
+instance[2] = 42;
+value = instance[2];
+delete instance[2];
```

Comment:
please can you explain how `delete` is implemented?  EG does this mean that you could have two objects of the same class, but one has `undefined` for the property because it has been deleted but the other does not, even if the definition of the property says that an uninitialised value is impossible?

Reply:
Calling `delete instance[2]` is identical to calling `instance.removeAt(2)` because the `delete` is trapped and `removeAt` is actually called in its stead.